components/CodeTabs/index.tsx

```diff
@@ -56,7 +56,13 @@ const CodeTabs = (props: Props) => {
     <div className={`CodeTabs CodeTabs_initial theme-${theme}`}>
       <div className="CodeTabs-toolbar">
         {(Array.isArray(children) ? children : [children]).map((pre, i) => {
-          const { meta, lang } = pre.props.children.props;
+          // Access lang/meta from the Code component's props
+          // pre.props.children is an array of Code components
+          const codeComponent = Array.isArray(pre.props?.children)
+            ? pre.props.children[0]
+            : pre.props?.children;
```

Comment:
Why is this different between MDX and MDXish? This comment could also be clarified; maybe something like:


```suggestion
          // the first or only child should be our Code component:
          const codeComponent = Array.isArray(pre.props?.children)
            ? pre.props.children[0]
            : pre.props?.children;
```

Reply:
- With how the MDXish pipeline processes the input slightly differently, I found cases where the props children is an array instead of just an object. It's just the plugins we've had to add may parse the props slightly differently
- I think this makes the code slightly more typesafe, since children is of any type and more type guards would be better in my opinion
- Clarified the comment!